concourse/scripts/kipper.sh

```diff
@@ -0,0 +1,358 @@
+  local level
+  level="${2:-FATAL}"
+  echo -e "${level}: ${string}" >&2
+  if [[ "${level}" = 'FATAL' ]]; then
+    exit 1
+  fi
+}
+
+check_dependencies() {
+  echo 'checking dependencies...'
+  local bash_ver
+  bash_ver=$(echo "${BASH_VERSION}" | cut -d'.' -f1)
+  if [[ "${bash_ver}" -lt 4 ]]; then
+    print_error 'requires bash version >= 4'
+  fi
+  if ! command -v bosh >/dev/null 2>&1; then
+    print_error 'missing bosh cli'
+  fi
+  if ! command -v jq >/dev/null 2>&1; then
+    print_error 'missing jq'
+  fi
+  if ! command -v aws >/dev/null 2>&1; then
+    print_error 'missing aws cli'
+  fi
+}
+
+validate_initial_state() {
+  echo 'validating initial state...'
+  if ! bosh vms >/dev/null 2>&1; then
+    print_error 'not logged in to bosh, or cf not deployed'
+  fi
+  # make sure we are logged into the correct bosh env
+  if [[ "${BOSH_ENVIRONMENT:?}" != "bosh.${env_arg}.dev.cloudpipeline.digital" ]]; then
+    print_error "wrong bosh env targeted (expecting: ${env_arg})"
+  fi
+  if ! aws s3 ls >/dev/null 2>&1; then
+    print_error 'not logged in to aws'
+  fi
+  local ec2_instances
+  ec2_instances=$(get_ec2_instances)
+  local ec2_instances_ids
+  ec2_instances_ids=$(get_ec2_instances_ids "${ec2_instances}")
+  local ec2_instances_count
+  ec2_instances_count=$(get_length "${ec2_instances_ids}")
+  if [[ "${ec2_instances_count}" = '0' || "${ec2_instances_count}" = '' ]]; then
+    print_error "env does not exist: ${env_arg}"
+  fi
+  # check that ec2 instances are not already mid state change
+  local ec2_instances_transitioning
+  ec2_instances_transitioning=$(get_ec2_instances 'stopping,starting,pending,shutting-down')
+  local ec2_instances_transitioning_ids
+  ec2_instances_transitioning_ids=$(get_ec2_instances_ids "${ec2_instances_transitioning}")
+  local ec2_instances_transitioning_count
+  ec2_instances_transitioning_count=$(get_length "${ec2_instances_transitioning_ids}")
+  if [[ "${ec2_instances_transitioning_count}" != '0' || "${ec2_instances_transitioning_count}" = '' ]]; then
+    print_error "env is already in transition: ${env_arg}"
+  fi
+}
+
+parse_args() {
+  env_arg="${1:-}"
+  if [[ "${env_arg}" = '' ]]; then
+    print_error 'expecting valid env in arg1 ( eg: dev01 )'
+  fi
+  action_arg="${2:-}"
+  if ! [[ "${action_arg}" = 'kip' || "${action_arg}" = 'wake' ]]; then
+    print_error 'expecting valid action in arg2 ( kip | wake )'
+  fi
+  if [[ "${action_arg}" = 'kip' ]]; then
+    action='stop'
+    initial_ec2_state='running'
+    target_ec2_state='stopped'
+    initial_db_state='available'
+    target_db_state='stopped'
+    past_tense='abed'
+  elif [[ "${action_arg}" = 'wake' ]]; then
+    action='start'
+    initial_ec2_state='stopped'
+    target_ec2_state='running'
+    initial_db_state='stopped'
+    target_db_state='available'
+    past_tense='awake'
+  fi
+}
+
+print_header() {
+  echo '
+
+                  o
+  |/. _  _  _  _  .
+  |\||_)|_)(/_|    <><
+     |  |          
+  '
```

Comment:
The cute name's probably ok, but I'd suggest it _also_ says what it actually does somewhere.

Also, I think the "abed" thing is a bit too far. The sentence doesn't even really parse for me.

Reply:
oh i thought you'd like that - changed to the more conventional "asleep"

Reply:
Yeah, but it's not a matter of what I'd like, I don't think the whole kipper thing means anything to non-native-english speakers.